plugin/pkg/up/up.go

```diff
@@ -37,9 +38,11 @@ func (p *Probe) Stop() { p.stop <- true }
 func (p *Probe) Start(interval time.Duration) { go p.start(interval) }
 
 func (p *Probe) start(interval time.Duration) {
+	stop := uint32(0)
 	for {
 		select {
 		case <-p.stop:
+			atomic.StoreUint32(&stop, 1)
```

Comment:
this is purely for testing?

Reply:
Actually not. The test run successfully without this change. The goroutine below is stopped at the end of test and doesn't affect results
This is rather a fix of probe logic. If we call `Probe.Stop()` we expect it will stop probing. But before my fix in case if probing function always return error it keeps probing forever, which seems incorrect